kythe/go/platform/tools/indexer_wrapper/local.go

```diff
@@ -0,0 +1,99 @@
+// This package provides an implementation of the CompilationAnalyzer interface
+// that works by saving the compilation unit to a temporary kzip file and
+// invoking an indexer binary on it.
+package main
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+
+	"kythe.io/kythe/go/platform/analysis"
+	"kythe.io/kythe/go/platform/delimited"
+	"kythe.io/kythe/go/platform/kzip"
+	apb "kythe.io/kythe/proto/analysis_go_proto"
+)
+
+type localAnalyzer struct {
+	cmd     []string
+	fetcher analysis.Fetcher
+}
+
+// NewLocalAnalyzer creates an analyzer that will execute the given command when
+// Analyze() is called. The filepath to a kzip is appended to the command, so if
+// given ["my_indexer", "--index"], the final executed command will be
+// ["my_indexer", "--index", "some/file.kzip"].
+func NewLocalAnalyzer(cmd []string, fetcher analysis.Fetcher) *localAnalyzer {
+	return &localAnalyzer{cmd: cmd, fetcher: fetcher}
+}
+
+func (a *localAnalyzer) Analyze(ctx context.Context, req *apb.AnalysisRequest, f analysis.OutputFunc) error {
+	tmpkzip, err := saveSingleUnitKzip(ctx, req.Compilation, a.fetcher)
+	if err != nil {
+		return fmt.Errorf("creating tmp kzip: %v", err)
+	}
+	defer os.Remove(tmpkzip)
+
+	// Run indexer binary
+	var errBuf strings.Builder
+	var outBuf bytes.Buffer
+	cmd := exec.Command(a.cmd[0], append(a.cmd[1:], tmpkzip)...)
```

Comment:
Check that `cmd` has the proper shape in `NewLocalAnalyzer` (adding a new error return).